crates/ruma-api-macros/src/util.rs

```diff
@@ -77,3 +77,32 @@ pub fn extract_cfg(attr: &Attribute) -> Option<NestedMeta> {
 
     Some(list.nested.pop().unwrap().into_value())
 }
+
+pub fn convert_path_string(
```

Comment:
I think this should have a more specific name and include the `quote!` that's currently in the "mapper" closure that calls this. Suggested name:

```suggestion
pub fn path_format_args_call(
```

Reply:
That quote isn't possible, it requires `percent_encoding`, thats why there's a mapper function in the first place, to make the `.map` operation less noisy